messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/ExternalPooledConnectionFactoryAdd.java

```diff
@@ -0,0 +1,206 @@
+
+import org.jboss.as.connector.metadata.deployment.ResourceAdapterDeployment;
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.naming.deployment.ContextNames;
+import org.jboss.as.naming.deployment.ContextNames.BindInfo;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.wildfly.extension.messaging.activemq.CommonAttributes;
+import org.wildfly.extension.messaging.activemq.DiscoveryGroupDefinition;
+import org.wildfly.extension.messaging.activemq.MessagingServices;
+import org.wildfly.extension.messaging.activemq.TransportConfigOperationHandlers;
+import org.wildfly.extension.messaging.activemq.jms.ConnectionFactoryAttributes.Common;
+
+/**
+ *
+ * @author Emmanuel Hugonnet (c) 2018 Red Hat, inc.
+ */
+public class ExternalPooledConnectionFactoryAdd extends AbstractAddStepHandler {
+
+    public static final ExternalPooledConnectionFactoryAdd INSTANCE = new ExternalPooledConnectionFactoryAdd();
+
+    private ExternalPooledConnectionFactoryAdd() {
+        super(getDefinitions(ExternalPooledConnectionFactoryDefinition.ATTRIBUTES));
+    }
+
+    @Override
+    protected void performRuntime(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
+
+        ModelNode model = resource.getModel();
+        PathAddress address = context.getCurrentAddress();
+        final String name = context.getCurrentAddressValue();
+
+        final ModelNode resolvedModel = model.clone();
+        for(final AttributeDefinition attribute : getDefinitions(ExternalPooledConnectionFactoryDefinition.ATTRIBUTES)) {
+            resolvedModel.get(attribute.getName()).set(attribute.resolveModelAttribute(context, resolvedModel ));
+        }
+
+        // We validated that jndiName part of the model in populateModel
+        final List<String> jndiNames = new ArrayList<String>();
+        for (ModelNode node : resolvedModel.get(Common.ENTRIES.getName()).asList()) {
+            jndiNames.add(node.asString());
+        }
+        final BindInfo bindInfo = ContextNames.bindInfoFor(jndiNames.get(0));
+        List<String> jndiAliases;
+        if(jndiNames.size() > 1) {
+            jndiAliases = new ArrayList<>(jndiNames.subList(1, jndiNames.size()));
+        } else {
+            jndiAliases = Collections.emptyList();
+        }
+        String managedConnectionPoolClassName = null;
+        if (resolvedModel.hasDefined(ConnectionFactoryAttributes.Pooled.MANAGED_CONNECTION_POOL.getName())) {
+            managedConnectionPoolClassName = resolvedModel.get(ConnectionFactoryAttributes.Pooled.MANAGED_CONNECTION_POOL.getName()).asString();
```

Comment:
asStringOrNull

No need to update this for my 'OrNull' comments unless you are making other changes.